Simulation initialization fail for some seed with negative coin amount

To reproduce: use `seed=54` for `make test-sim-ci`

The initialization of the simulation fails with the error

```
Running application benchmark for numBlocks=200, blockSize=26
Starting SimulateFromSeed with randomness created with seed 54
Randomized simulation params:
{}
Selected randomly generated parameters for simulated genesis:
{
  stake_per_account: "4090889436331169",
  initially_bonded_validators: "352"
}
Selected randomly generated auth parameters:
{
 "max_memo_characters": 149,
 "tx_sig_limit": 7,
 "tx_size_cost_per_byte": 5,
 "sig_verify_cost_ed25519": 570,
 "sig_verify_cost_secp256k1": 936
}
panic: negative coin amount: -6910435863255655036

goroutine 66 [running]:
github.com/cosmos/cosmos-sdk/types.NewCoin({0x56eadb6, 0x5}, {0x90c31})
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/types/coin.go:23 +0x51
github.com/cosmos/cosmos-sdk/x/bank/simulation.RandomizedGenState(0xc00057f7c0)
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/x/bank/simulation/genesis.go:69 +0x20c
github.com/cosmos/cosmos-sdk/x/bank.AppModule.GenerateGenesisState(...)
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/x/bank/module.go:175
github.com/cosmos/cosmos-sdk/types/module.(*SimulationManager).GenerateGenesisStates(0x60b45a0, 0xc000010018)
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/types/module/simulation.go:72 +0x6b
github.com/tendermint/spn/app/simutil.CustomAppStateFn.func1(0xc000ee0bd0, {0xc000e12000, 0x9a4, 0x2}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...})
	/Users/lucas/go/src/github.com/tendermint/spn/app/simutil/app_state.go:71 +0x3e5
github.com/cosmos/cosmos-sdk/x/simulation.initChain(0xc00057edc0, {0x3fb1c85f6ea183cc, 0x9a4, 0x3f9695535acaa024, {0xc000616ee8, _, _}, {_, _}, {_, ...}}, ...)
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/x/simulation/simulate.go:35 +0xce
github.com/cosmos/cosmos-sdk/x/simulation.SimulateFromSeed({0x6126360, 0xc000e7e000}, {0x60b45a0, 0xc000010018}, 0x0, _, _, {0xc00052aa00, 0x37, 0x50}, ...)
	/Users/lucas/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.2/x/simulation/simulate.go:77 +0x4d0
github.com/tendermint/spn/app_test.BenchmarkSimulation(0xc000e7e000)
	/Users/lucas/go/src/github.com/tendermint/spn/app/simulation_test.go:89 +0x625
testing.(*B).runN(0xc000e7e000, 0x1)
	/usr/local/go/src/testing/benchmark.go:192 +0x126
testing.(*B).run1.func1()
	/usr/local/go/src/testing/benchmark.go:232 +0x59
created by testing.(*B).run1
	/usr/local/go/src/testing/benchmark.go:225 +0xa5
```

@aljo242 this seems to come from `CustomAppStateFn` from https://github.com/tendermint/spn/pull/646
It may potentially come from a unsigned int conversion, we have
```
var initialStake, numInitiallyBonded int64
appParams.GetOrGenerate(
	cdc, simappparams.StakePerAccount, &initialStake, r,
	func(r *rand.Rand) { initialStake = r.Int63n(1e16) },
)
appParams.GetOrGenerate(
	cdc, simappparams.InitiallyBondedValidators, &numInitiallyBonded, r,
	func(r *rand.Rand) { numInitiallyBonded = int64(r.Intn(300) + 200) },
)
```

--- a/app/simulation_test.go
+++ b/app/simulation_test.go
@@ -52,7 +52,6 @@ func interBlockCacheOpt() func(*baseapp.BaseApp) {
 
 // go test ./app -v -benchmem -run=^$ -bench ^BenchmarkSimulation -Commit=true -cpuprofile cpu.out
 func BenchmarkSimulation(b *testing.B) {
-	simapp.FlagSeedValue = 10
 	simapp.FlagVerboseValue = true
 	simapp.FlagOnOperationValue = true
 	simapp.FlagAllInvariantsValue = true

--- a/app/simutil/app_state.go
+++ b/app/simutil/app_state.go
@@ -34,14 +34,14 @@ func CustomAppStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager)
 		chainID = config.ChainID
 		appParams := make(simtypes.AppParams)
 
-		maxInitialStake := math.MaxInt64 / (numAccs + MaxNumBonded)
+		maxInitialCoin := math.MaxInt64 / (numAccs + MaxNumBonded)
 
 		// generate a random amount of initial stake coins and a random initial
 		// number of bonded accounts
 		var initialStake, numInitiallyBonded int64
 		appParams.GetOrGenerate(
 			cdc, simappparams.StakePerAccount, &initialStake, r,
-			func(r *rand.Rand) { initialStake = r.Int63n(maxInitialStake) },
+			func(r *rand.Rand) { initialStake = r.Int63n(maxInitialCoin) },
 		)
 		appParams.GetOrGenerate(
 			cdc, simappparams.InitiallyBondedValidators, &numInitiallyBonded, r,
@@ -119,7 +119,7 @@ func CustomAppStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager)
 		totalNewCoins := sdk.NewCoins()
 		for i, balance := range bankState.Balances {
 			if r.Int63n(100) < 20 {
-				auctionCoinAmt := r.Int63n(999_000_000_000_000) + 1_000_000_000_000
+				auctionCoinAmt := r.Int63n(maxInitialCoin)
 				auctionCoin := sdk.NewCoin(AuctionCoinDenom, sdk.NewInt(auctionCoinAmt))
 				newBalance := balance.Coins.Add(auctionCoin)
 				bankState.Balances[i].Coins = newBalance

--- a/app/simutil/app_state.go
+++ b/app/simutil/app_state.go
@@ -3,6 +3,7 @@ package simutil
 import (
 	"encoding/json"
 	"fmt"
+	"math"
 	"math/rand"
 	"time"
 
@@ -19,6 +20,7 @@ import (
 
 const (
 	AuctionCoinDenom = "auction"
+	MaxNumBonded     = 300
 )
 
 // CustomAppStateFn returns the initial application state using the simulation parameters.
@@ -32,16 +34,18 @@ func CustomAppStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager)
 		chainID = config.ChainID
 		appParams := make(simtypes.AppParams)
 
+		maxInitialStake := math.MaxInt64 / (numAccs + MaxNumBonded)
+
 		// generate a random amount of initial stake coins and a random initial
 		// number of bonded accounts
 		var initialStake, numInitiallyBonded int64
 		appParams.GetOrGenerate(
 			cdc, simappparams.StakePerAccount, &initialStake, r,
-			func(r *rand.Rand) { initialStake = r.Int63n(1e16) },
+			func(r *rand.Rand) { initialStake = r.Int63n(maxInitialStake) },
 		)
 		appParams.GetOrGenerate(
 			cdc, simappparams.InitiallyBondedValidators, &numInitiallyBonded, r,
-			func(r *rand.Rand) { numInitiallyBonded = int64(r.Intn(300) + 200) },
+			func(r *rand.Rand) { numInitiallyBonded = r.Int63n(MaxNumBonded) },
 		)
 
 		if numInitiallyBonded > numAccs {
@@ -52,9 +56,10 @@ func CustomAppStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager)
 			`Selected randomly generated parameters for simulated genesis:
 {
   stake_per_account: "%d",
-  initially_bonded_validators: "%d"
+  initially_bonded_validators: "%d",
+  num_accs: "%d"
 }
-`, initialStake, numInitiallyBonded,
+`, initialStake, numInitiallyBonded, numAccs,
 		)
 
 		simState := &module.SimulationState{

